### PR TITLE
Nested tables

### LINKED ISSUE #154: [SVG] table lines are not showing up in IE
It looks like the `outline` CSS that works everywhere else doesn't work in IE, so these will have to be handled explicitly by the SVG renderer rather than falling back on CSS styles.

### PR DESCRIPTION
**Pull request by dpvc**
This PR adds better support for nested tables, and in particular, nested tables with labels and with percentage widths.  The percentage-width tables are taken to be a percentage of the containing table cell (the line-wrap context, according to the MathML spec).  Nesting is also considered for fractions and `mpadded` elements.  An extra `data-align` attribute can be used to control the alignment in `mpadded` elements; the `columnalign` attribute controls alignment in tables nested inside tables, and `numalign`/`denomalign` control it in tables nested in fractions.

This PR also adds 60 test files (so more than half the files are just those tests).

Among other things, this resolves #154, #163, #164, #183, #184, and #185.

### CODE REVIEW

File: mathjax3-ts/output/svg/Wrappers/mtable.ts [hunk 1]
```diff
@@ -191,118 +236,95 @@ CommonMtableMixin<SVGmtd<N, T, D>, SVGmtr<N, T, D>, SVGConstructor<N, T, D>>(SVG
...
-            'data-frame': true,
-            width: this.fixed(w), height: this.fixed(h+d), y: this.fixed(-d)
-        }
-        if (style !== 'solid') {
-            properties.style = {'outline-style': style};
-        }
-        return this.svg('rect', properties);
+        const t = this.fLine;
+        return this.svg('rect', this.setLineThickness(t, style, {
+            'data-frame': true, 'class': 'mjx-' + style,
```
Comment (zorkow) [inline mathjax3-ts/output/svg/Wrappers/mtable.ts]: Is there no method for adding the prefix? Or a constant that defines the prefix? Just in the event we ever want to change it in the future.

File: samples/tables/lib/dual-tables.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+const div = document.createElement('div');
+const template = document.getElementsByTagName('mjx-template')[0];
+
+function  CreateMathML() {
+    const mathml = template.innerHTML;
+    template.innerHTML = mathml.replace(/\&/g, '&amp;')
+                               .replace(/</g, '&lt;')
+                               .replace(/>/g, '&gt;')
+                               .replace(/\n/g, '<br/>')
+                               .replace(/\s/g, '\u00A0');
```
Comment (zorkow) [inline samples/tables/lib/dual-tables.js]: Any chance this might ever change or grow? Then it might be easier to put it into a pair list that's processed.

File: samples/tables/lib/dual-tables.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    Substitute(mathml, window.variables, '');
+}
+
+function Substitute(mml, data, descr) {
+  if (data.length === 0) {
+    addMathML(mml, descr);
+  } else {
```
Comment (zorkow) [inline samples/tables/lib/dual-tables.js]: Return after the corner case.

File: samples/tables/lib/dual-tables.js (hunk 2 above)
Comment (dpvc) [reply samples/tables/lib/dual-tables.js]: I don't think so.  This is to format the original HTML source from the template so that it is displayed as code in the browser.  These are the things that need to be changed to do that (HTML specials, newlines and spaces).   I suppose the spaces and newlines could be handed by setting the CSS for white-space, but otherwise, this should not need to be changed.

File: mathjax3-ts/output/common/Wrappers/math.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ * @template T  The Wrapper class constructor type
+ */
+export function CommonMathMixin<T extends WrapperConstructor>(Base: T): MathConstructor & T {
+    return class extends Base {
+
+        /**
+         * @override
+         */
+        public getWrapWidth(i: number) {
+            return (this.parent ? this.getBBox().w : this.metrics.containerWidth / this.jax.pxPerEm);
```
[the remark this answers is not in the document]
Comment (dpvc) [reply mathjax3-ts/output/common/Wrappers/math.ts]: We agreed that this could be allowed (some time ago) and it appears all over the place in the code.  If we want to remove them, let's add that to the updates issue and handle it in one edit.
